apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/SubscriberClientTest.cs

```diff
@@ -66,6 +67,32 @@ private ServerAction(TimeSpan preInterval, IEnumerable<string> msgs, RpcExceptio
             public int? DeliveryAttempt { get; }
         }
 
+        // Class to simulate exceptions in Acknowledgement and/or ModifyAcknowledgementDeadline calls.
+        private class AckModifyAckDeadlineAction
+        {
+            public static AckModifyAckDeadlineAction Data(Exception ex, int numberOfFailures, bool onAck, bool onModifyAckDeadline) => new AckModifyAckDeadlineAction(ex, numberOfFailures, onAck, onModifyAckDeadline);
+
+            public static AckModifyAckDeadlineAction BadAck(Exception ex, int numberOfFailures) => new AckModifyAckDeadlineAction(ex, numberOfFailures, true, false);
+            
+            public static AckModifyAckDeadlineAction BadModifyAckDeadline(Exception ex, int numberOfFailures) => new AckModifyAckDeadlineAction(ex, numberOfFailures, false, true);
+            
+            private AckModifyAckDeadlineAction(Exception exception, int numberOfFailures, bool onAck, bool onModifyAckDeadline)
+            {
+                Exception = exception;
+                NumberOfFailures = numberOfFailures;
+                OnAck = onAck;
+                OnModifyAckDeadline = onModifyAckDeadline;
+            }
+
+            public Exception Exception { get; }
```

Comment:
Nit: please move the properties to above the constructor. I'd also remove the blank lines between the properties, and between the three factory methods. (But keep a line between the three factory methods and the properties.)

Reply:
Done.